parsers/NTESMO.py

```diff
@@ -71,90 +76,112 @@ class Generator(TypedDict):
     status_forcelist=[500, 502, 503, 504],
 )
 
-_DT_CLASS = "smp-tiles-article__title"
 
+def _scan_daily_report_urls(
+    urls: Iterable[str], session: Session
+) -> Iterator[tuple[date, str]]:
+    """Scans pages of daily report cards yielding links to the excel data files for each day."""
+
+    for url in urls:
+        response = session.get(url)
+        if not response.ok:
+            raise ParserException(
+                PARSER,
+                f"Exception when fetching daily trading data error code: {response.status_code}: {response.text}",
+            )
+
+        page = response.text
+        soup = BeautifulSoup(page, "html.parser")
+        for a in soup.find_all("a", href=True):
+            if a["href"].startswith("https://ntesmo.com.au/__data/assets/excel_doc/"):
+                timestamp = a.find("div", {"class": "smp-tiles-article__title"}).text
+                dt = (
+                    datetime.strptime(timestamp, "%d %B %Y")
+                    .replace(tzinfo=AUSTRALIA_TZ)
+                    .date()
+                )
```

Comment:
When using `--target_datetime=2018-03-11` for example this returns multiple strings of dates which is causing the date parsing to fail.

Reply:
Looks like there are a bunch of non-standard cards on the website whose `article__title` has some text in addition to the data or e.g. have malformatted dates  👀 

```
# examples
"05 April 2018 - Publishing Recommencement"
"05 January 2018 - System Notice"
"18 June2018"
```

I'll try to go through all years and see if I can implement some parsing logic that can shield us from all of those

Reply:
Ok - I have updated the code to be able to handle all (known) special cases!